HTML-CSS-JS/html-css.md

```diff
@@ -264,196 +264,272 @@ If you have 4 elements, then you will get two rows and three columns. in the fir
 
-1
-HTML
+---
 
+## Question 15
+### 1.15 What is an iframe and how does it work? Medium
+`iframe`  - another HTML element inside a web page.
+They are commonly used to external ads, videos, tags, or other interactive elements into the page.
 
+---
 
-13
-1.13
-Ways to improve website performance
-Medium
+## Question 16
+### 1.16 Explain the use of `rel="nofollow"`, `rel="noreferrer"`, `rel="noopener"` attribute? Medium
 
+`rel="noreferrer"` keeps external sites from knowing that you have included links to their material on your website
 
-14
-1.14
-What does async and defer refer in script tag? Describe the difference between <script>, <script async> and <script defer>
+`rel="nofollow"` that directs search engines not to use the link for page ranking calculations.
 
-Medium
+`rel="noopener"`  - browser navigate to the target resource without granting the new browsing context access to the document that opened it
 
+---
 
-15
-1.15
-What is an iframe and how does it work?
-Medium
+## Question 17
+### 1.17 Explain the use of `rel="preload"`, `rel="prefetch"`, `rel="preconnect"` attribute? Medium
+`rel="preload"` - the browser should download and cache a resource as soon as possible
 
+`rel="prefetch"` -the browser should download and cache a resource in the background
 
-16
-1.16
-Explain the use of `rel="nofollow"`, `rel="noreferrer"`, `rel="noopener"` attribute?
-Medium
+`rel="preconnect"`  - the browser should download a resource as soon as possible because your page will set up a connection to another origin
 
+---
 
-17
-1.17
-Explain the use of `rel="preload"`, `rel="prefetch"`, `rel="preconnect"` attribute?
-Medium
+## Question 18
+### 1.18 What does CORS stand for and what issue does it address? Medium
 
+CORS stands for Cross Origin Resource Sharing and is used to get around the browsers same-origin policy. For security purposes, a browser won't load requests for resources to other domains when those requests are initiated by scripts. CORS gets around this issue by supplying a special header that specifies which domains may make XMLHttpRequests for its resources.
 
-18
-1.18
-What does CORS stand for and what issue does it address?
-Medium
+---
 
+## Question 19
+### 1.19 What is the DOM? How does the DOM work? Medium
+The DOM is an interface(API) that allows a programming language to manipulate the structure, the content and style of a website. The main object “document” has properties, methods, and events available for manipulating
 
-19
-1.19
-What is the DOM? How does the DOM work?
-Medium
+---
 
+## Question 33
+### 2.10 What are the css selectors? Medium
+`p {color: red}`
+A CSS selector is the first part of a CSS Rule. In ours example it is `“p”`
+CSS selectors can be tags(`p,a,img,div,body`), class(`.box, .container`), id (`#targetElement`)
 
+You can combine different types
+Body `div.contaner p {color: blue}`
 
-2
-CSS
+---
 
+## Question 34
+### 2.11 When to use css grid and flexbox? Medium
```

Comment:
Not exactly, I can set up direction for flex-box and put elements on X or Y axis. Please, add more details